other_procore_apis/bim_web_viewer.md

```diff
@@ -832,13 +791,541 @@ Manually dispatches an event
+
+---
+
+### isolateCompleted
+
+<p class="heading-link-container"><a class="heading-link" href="#isolatecompleted"></a></p>
+
+Fires when the `Isolate` from the right click context menu operation has completed.
+
+---
+
+### messages
+
+<p class="heading-link-container"><a class="heading-link" href="#messages"></a></p>
+
+Fires when the webviewer throws a warning or error.
+
+#### Data Properties
+
+| Field Name  | Type   | Description                                       |
+| ----------- | ------ | ------------------------------------------------- |
+| messageType | string | The type of message, either `Warning` or `Error`. |
+| component   | string | Where the message originates from.                |
+| message     | string | A human readable message.                         |
+
+---
+
+### measured
+
+<p class="heading-link-container"><a class="heading-link" href="#measured"></a></p>
+
+Fires when measuring two objects has completed.
+
+---
+
+### modelTreeParentInteracted
+
+<p class="heading-link-container"><a class="heading-link" href="#modeltreeparentinteracted"></a></p>
+
+Fires when an object in the `Model Objects` has been interacted with.
+
+#### Data Properties
+
+| Field Name | Type   | Description                                                                                                                                                                   |
+| ---------- | ------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| children   | number | The number of children the interacted object has.                                                                                                                             |
+| action     | string | The type of action, can be the following: `checkbox`, `open`, `click`, `double_click` or `close`                                                                              |
+| depth      | number | The depth of the object interacted with. A depth of 0 indicates top level objects. A depth of 1 indicates an object that is one level down from root level objects and so on. |
+
+---
+
+### markupDisplayed
+
+<p class="heading-link-container"><a class="heading-link" href="#markupdisplayed"></a></p>
+
+Fires when markup from a viewpoint has been displayed.
+
+---
+
+### markupDismissed
+
+<p class="heading-link-container"><a class="heading-link" href="#markupdismissed"></a></p>
+
+Fires when markup from a viewpoint has been dismissed.
+
+---
+
+### navigationChanged
+
+<p class="heading-link-container"><a class="heading-link" href="#navigationchanged"></a></p>
+
+Fires when the navigation mode has changed.
+
+#### Data Properties
+
+| Field Name | Type   | Description                                     |
+| ---------- | ------ | ----------------------------------------------- |
+| mode       | number | The type of navigation that was just activated. |
```

Comment:
Guessing we're not documenting this specifically because it just gives you 0, 1, 2, 3 ? haha

I think that's coming from here and we don't expose it currently: https://github.com/procore/bim-webviewer/blob/develop/src/controllers/RenderWrapperController.js#L156-L162

Reply:
Yup, that is correct lol.

Reply:
We can update documentation if we decide to expose that enum.